src/main/java/de/muenchen/dave/services/messstelle/auswertung/AuswertungService.java

```diff
@@ -1,55 +1,118 @@
+            throw new IllegalArgumentException("Es wurden keine Messstellen ausgewählt.");
         }
-        return result;
+        final var auswertungenMqByMstId = this.ladeAuswertungGroupedByMstId(options);
+        return spreadsheetService.createFile(auswertungenMqByMstId, options);
+    }
+
+    protected List<AuswertungProMessstelle> ladeAuswertungGroupedByMstId(final MessstelleAuswertungOptionsDTO options) {
+
+        final List<Zeitraum> zeitraeume = this.createZeitraeume(options.getZeitraum(), options.getJahre());
+
+        final ConcurrentMap<String, List<AuswertungProMessstelleUndZeitraum>> auswertungenGroupedByMstId = CollectionUtils
+                .emptyIfNull(options.getMessstelleAuswertungIds())
+                .parallelStream()
+                .flatMap(messstelleAuswertungIdDTO -> CollectionUtils.emptyIfNull(zeitraeume)
+                        .parallelStream()
+                        .map(zeitraum -> {
+                            final var tagesaggregate = messwerteService.ladeTagesaggregate(options.getTagesTyp(), messstelleAuswertungIdDTO.getMqIds(),
+                                    zeitraum);
+                            return auswertungMapper.tagesaggregatDto2AuswertungProMessstelleUndZeitraum(tagesaggregate,
+                                    zeitraum, messstelleAuswertungIdDTO.getMstId());
+                        }))
+                .collect(Collectors.groupingByConcurrent(AuswertungProMessstelleUndZeitraum::getMstId));
+        return convertAuswertungen(auswertungenGroupedByMstId);
+    }
+
+    protected List<Zeitraum> createZeitraeume(final List<AuswertungsZeitraum> auswertungszeitraeume, final List<Integer> jahre) {
+        return ListUtils.emptyIfNull(auswertungszeitraeume)
+                .stream()
+                .flatMap(auswertungsZeitraum -> ListUtils.emptyIfNull(jahre)
+                        .stream()
+                        .map(jahr -> new Zeitraum(
+                                YearMonth.of(jahr, auswertungsZeitraum.getZeitraumStart().getMonth()),
+                                YearMonth.of(jahr, auswertungsZeitraum.getZeitraumEnd().getMonth()),
+                                auswertungsZeitraum)))
+                .toList();
     }
 
-    protected List<List<LocalDate>> calculateZeitraeume(final List<AuswertungsZeitraum> auswertungszeitraeume, final List<Integer> jahre) {
-        final List<List<LocalDate>> result = new ArrayList<>();
-        auswertungszeitraeume.forEach(auswertungszeitraum -> jahre.forEach(jahr -> {
-            LocalDate end = auswertungszeitraum.getZeitraumEnd().withYear(jahr);
-            if (AuswertungsZeitraum.FEBRUAR == auswertungszeitraum && jahr % 4 == 0) {
-                end = end.withDayOfMonth(29);
-            }
-            result.add(List.of(auswertungszeitraum.getZeitraumStart().withYear(jahr), end));
-        }));
-        return result;
+    protected List<AuswertungProMessstelle> convertAuswertungen(
```

Comment:
Den Methodennamen anpassen, so das dieser der fachlichen Tätigkeit entspricht die in dieser Methode ausgeführt wird.

Reply:
Angepasst, aber wenn dir was besseres einfällt gerne vorschlagen